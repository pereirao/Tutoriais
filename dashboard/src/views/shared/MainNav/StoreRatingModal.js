// @flow

import React from 'react';
// import _ from 'lodash';
import classNames from 'classnames/bind';
import I18n from '../../../localization';
import {session_helpers} from '../../../business/session';
import MBModal from '../../elements/MBModal';
import style_sheet from './MainNav.module.css';

const {ratingClass} = session_helpers;
const cx = classNames.bind(style_sheet);

type StoreRatingModalProps = { rating: number, hidden: boolean, close: () => void };
const StoreRatingModal = ({rating, hidden, close}: StoreRatingModalProps) => {
  if (hidden) return null;

  const rating_class = ratingClass(rating);

  return (
    <MBModal close={close}>
      <div className={style_sheet.storeRatingModalWrapper}>
        <h3 className={style_sheet.storeRatingTitle}>{I18n.t('ui.modal.store_rating.title')}</h3>
        <p className={style_sheet.storeRatingBody}>{I18n.t('ui.modal.store_rating.message')}</p>
        <div className={style_sheet.modalRatingWrapper}>
          <h3 className={cx('storeRatingLarge', rating_class)}>{rating}</h3>
          <h3 className={cx('storeRatingSubtext', rating_class)}>{I18n.t(`ui.modal.store_rating.${rating_class}`)}</h3>
        </div>
        <p className={style_sheet.storeRatingBody}>{I18n.t('ui.modal.store_rating.goal')}</p>
      </div>
    </MBModal>
  );
};

export default StoreRatingModal;
