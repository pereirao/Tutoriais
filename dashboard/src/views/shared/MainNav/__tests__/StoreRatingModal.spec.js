import React from 'react';

import StoreRatingModal from '../StoreRatingModal';

it('renders good correctly', () => {
  expect(shallow(<StoreRatingModal rating="4.9" />)).toMatchSnapshot();
});

it('renders meh correctly', () => {
  expect(shallow(<StoreRatingModal rating="4.6" />)).toMatchSnapshot();
});

it('renders bad correctly', () => {
  expect(shallow(<StoreRatingModal rating="3.1" />)).toMatchSnapshot();
});
