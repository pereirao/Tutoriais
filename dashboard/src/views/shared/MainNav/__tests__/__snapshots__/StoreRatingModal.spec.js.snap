// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bad correctly 1`] = `
<MBModal
  closeable={true}
  size="medium"
>
  <div
    className="storeRatingModalWrapper"
  >
    <h3
      className="storeRatingTitle"
    >
      Store Rating
    </h3>
    <p
      className="storeRatingBody"
    >
      Your score is influenced by factors like customer ratings, late deliveries, out-of-stock problems.
    </p>
    <div
      className="modalRatingWrapper"
    >
      <h3
        className="storeRatingLarge bad"
      >
        3.1
      </h3>
      <h3
        className="storeRatingSubtext bad"
      >
        NEEDS IMPROVEMENT
      </h3>
    </div>
    <p
      className="storeRatingBody"
    >
      Our best partners maintain a score of 4.75 or above. To get your score up, confirm and deliver orders in a timely manner and keep an accurate inventory to avoid stock issues.
    </p>
  </div>
</MBModal>
`;

exports[`renders good correctly 1`] = `
<MBModal
  closeable={true}
  size="medium"
>
  <div
    className="storeRatingModalWrapper"
  >
    <h3
      className="storeRatingTitle"
    >
      Store Rating
    </h3>
    <p
      className="storeRatingBody"
    >
      Your score is influenced by factors like customer ratings, late deliveries, out-of-stock problems.
    </p>
    <div
      className="modalRatingWrapper"
    >
      <h3
        className="storeRatingLarge good"
      >
        4.9
      </h3>
      <h3
        className="storeRatingSubtext good"
      >
        KEEP IT UP
      </h3>
    </div>
    <p
      className="storeRatingBody"
    >
      Our best partners maintain a score of 4.75 or above. To get your score up, confirm and deliver orders in a timely manner and keep an accurate inventory to avoid stock issues.
    </p>
  </div>
</MBModal>
`;

exports[`renders meh correctly 1`] = `
<MBModal
  closeable={true}
  size="medium"
>
  <div
    className="storeRatingModalWrapper"
  >
    <h3
      className="storeRatingTitle"
    >
      Store Rating
    </h3>
    <p
      className="storeRatingBody"
    >
      Your score is influenced by factors like customer ratings, late deliveries, out-of-stock problems.
    </p>
    <div
      className="modalRatingWrapper"
    >
      <h3
        className="storeRatingLarge meh"
      >
        4.6
      </h3>
      <h3
        className="storeRatingSubtext meh"
      >
        YOU CAN DO BETTER
      </h3>
    </div>
    <p
      className="storeRatingBody"
    >
      Our best partners maintain a score of 4.75 or above. To get your score up, confirm and deliver orders in a timely manner and keep an accurate inventory to avoid stock issues.
    </p>
  </div>
</MBModal>
`;
